components/EmotionForm.js

```diff
@@ -69,96 +70,312 @@ const StyledSubmitButton = styled.button`
   }
 `;
 
+const StyledLabel = styled.label`
+  position: relative;
+`;
+
+const ToggleSwitch = styled(Circle)`
+  width: 1.7rem;
+  position: absolute;
+  display: inline;
+  fill: ${({ $color }) => ($color ? "#00b400" : "#cc0100")};
+  margin-left: 0.5rem;
+  bottom: calc(50% - 0.85rem);
+`;
+
 export default function EmotionForm({
-  name,
-  color,
-  onAddEmotionDetails,
+  editMode,
+  onSubmit,
   id,
-  slug,
-  emotionEntries,
-  subemotions,
+  correspondingEntry,
 }) {
   const router = useRouter();
-  const correspondingEntry = emotionEntries.find((entry) => entry.id === id);
+
+  const {
+    date,
+    emotion,
+    tensionLevel,
+    subemotion,
+    intensity,
+    category,
+    trigger,
+    notes,
+  } = correspondingEntry;
+
+  // for controlled-inputs
+  const [formValues, setFormValues] = useState({
+    tensionValue: tensionLevel,
+    emotionValue: emotion,
+    selectedSubemotionValue: subemotion,
+    intensityValue: intensity ? intensity : 0,
+    categoryValue: category,
+    triggerValue: trigger,
+    notesValue: notes,
+    subemotionOptions: [],
+    colorValue: "",
+  });
+
+  // for enable/disable intensity/category inputs
+  const [toggleRangeInputs, setTogggleRangeInputs] = useState({
+    toggleIntensity: true,
+    toggleCategory: true,
+  });
+  const { toggleIntensity, toggleCategory } = toggleRangeInputs;
+
+  const [showConfirmMessage, setShowConfirmMessage] = useState(false);
+
+  useEffect(
+    () =>
+      setFormValues({
+        ...formValues,
+        subemotionOptions: subemotions,
+        colorValue: color,
+      }),
+
+    []
+  );
+
+  const {
+    tensionValue,
+    emotionValue,
+    selectedSubemotionValue,
+    intensityValue,
+    categoryValue,
+    triggerValue,
+    notesValue,
+    subemotionOptions,
+    colorValue,
+  } = formValues;
+
+  const inCaseOfNoEmotion = {
+    color: "lightgray",
+    subemotions: [],
+  };
+
+  const correspondingEmotion = emotion
+    ? emotionData.find((emotionObject) => emotionObject.name === emotion)
+    : inCaseOfNoEmotion;
+  const { subemotions, color } = correspondingEmotion;
```

Comment:
You should not use a useEffect here, since you only want fallback data for your "color" and "subemotions" values for the initial values of your form. You can pass a function to a useState, the return value of this function will be the initial state of your state. In that function you can perform these "choose emotionData or fallbackData" actions you perform here:
```js
const [formValues, setFormValues] = useState(() => {
  // peform any searching or sorting, renaming in here
  const {color, subemotions} = emotion
    ? emotionData.find((emotionObject) => emotionObject.name === emotion)
    : inCaseOfNoEmotion;
  return {
    //initial values for your form,
     subemotionOptions: subemotions,
     colorValue: color,
  }
})
```

Reply:
Thanks for your review we changed the code using your suggestion. Code is shorter and better now, thanks again!